File: tools/mcp-spec-audit/test/fixtures/clean-server/clean.py
"""Compliant MCP server fixture: the scanner should report zero findings."""

from mcp.server import MCPServer

mcp = MCPServer("clean-demo")


@mcp.tool()
def add(a: int, b: int) -> int:
    return a + b


@mcp.tool()
def read_notes(directory: str) -> str:
    # Scope arrives as an explicit tool parameter, not via Roots.
    return f"reading notes from {directory}"

File: tools/mcp-spec-audit/test/fixtures/clean-server/clean.ts
/* Compliant MCP server fixture: the scanner should report zero findings. */
import { MCPServer } from "@modelcontextprotocol/server";

const server = new MCPServer({ name: "clean-demo", version: "0.1.0" });

server.tool("add", { a: "number", b: "number" }, async ({ a, b }) => a + b);

export { server };
